std/machines/arith_bb.asm

```diff
@@ -0,0 +1,197 @@
+use std::array;
+use std::utils::unchanged_until;
+use std::utils::force_bool;
+use std::utils::sum;
+use std::math::ff;
+use std::check::panic;
+use std::convert::int;
+use std::convert::fe;
+use std::convert::expr;
+use std::prover::eval;
+use std::prelude::Query;
+use std::machines::range::Byte;
+
+// Arithmetic machine, ported mainly from Polygon: https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/arith.pil
+// Currently only supports "Equation 0", i.e., 256-Bit addition and multiplication.
+machine Arith16 with
+    latch: CLK8_7,
+    operation_id: operation_id,
+    // Allow this machine to be connected via a permutation
+    call_selectors: sel,
+{
+    Byte byte;
+    
+    // The operation ID will be bit-decomposed to yield selEq[], controlling which equations are activated.
+    col witness operation_id;
+
+    // operation_id has to be either mul or div.
+    operation_id * (1 - operation_id) = 0;
```

Comment:
You can also use std::utils::force_bool(operation_id); here

Reply:
Agree

Reply:
Done.